src/Sylius/Bundle/SettingsBundle/Manager/SettingsManager.php

```diff
@@ -71,29 +79,22 @@ class SettingsManager implements SettingsManagerInterface
      */
     protected $eventDispatcher;
 
-    /**
-     * @param SchemaRegistryInterface  $schemaRegistry
-     * @param ObjectManager            $parameterManager
-     * @param RepositoryInterface      $parameterRepository
-     * @param FactoryInterface         $parameterFactory
-     * @param Cache                    $cache
-     * @param ValidatorInterface       $validator
-     * @param EventDispatcherInterface $eventDispatcher
-     */
     public function __construct(
-        SchemaRegistryInterface $schemaRegistry,
-        ObjectManager $parameterManager,
-        RepositoryInterface $parameterRepository,
+        ServiceRegistryInterface $schemaRegistry,
+        ServiceRegistryInterface $resolverRegistry,
+        ObjectManager $settingsManager,
+        FactoryInterface $settingsFactory,
         FactoryInterface $parameterFactory,
-        Cache $cache,
+        SettingsResolverInterface $defaultResolver,
```

Comment:
That sounds like a great solution! But if people wanted to change the default resolver they would need to change the service that decorates the resolver registry service right?

What do you think about adding a configuration for the `sylius_settings` extension to easily change the service used as the default resolver in the decorator for the resolver registry? I think it might be a nice feature to have and easy to implement.